src/Components/SmartComponents/Modals/OsExposureReportModal/OsExposureReportModal.js

```diff
@@ -0,0 +1,234 @@
+import React, { useEffect, useState } from 'react';
+import propTypes from 'prop-types';
+import {
+    Modal,
+    Button,
+    Form,
+    FormGroup,
+    TextInput,
+    Flex,
+    FlexItem,
+    Checkbox,
+    TextArea
+} from '@patternfly/react-core';
+import LifecycleSelect from './LifecycleSelect';
+import isEqual from 'lodash/isEqual';
+import { CUSTOM_REPORT_FILTER_DATA_BY_LABEL, getCveReportFilters } from '../../../../Helpers/constants';
+
+const OsExposureReportModal = ({
+    isOsExposureModalOpen,
+    setOsExposureModalOpen,
+    osExposureData
+}) => {
+    const [reportTitle, setReportTitle] = useState('');
+    const [exposureMap, setExposureMap] = useState({});
+    const [selectedOsVersions, setSelectedOsVersions] = useState([]);
+    const [filterData, setFilterData] = useState({
+        impact: [],
+        advisory_available: ['true']
+    });
+    const [userNote, handleUserNoteChange] = useState('');
+    const CVE_REPORT_FILTERS = getCveReportFilters(false, true);
+    const ACTIVE_FILTERS = [
+        'impact',
+        'advisory_available'
+    ];
+
+    useEffect(() => {
+        if (osExposureData && osExposureData.length) {
+            const exposureMapCopy = { ...exposureMap };
+            osExposureData.sort((a, b) => {
+                if (a.major < b.major) {return 1;}
+
+                if (a.major > b.major) {return -1;}
+
+                if (a.minor < b.minor) {return 1;}
+
+                if (a.minor > b.minor) {return -1;}
```

Comment:
The same `if` conditions are repeated, also consider using lodash `sortBy`

Reply:
The if conditions are different. One set is for major and the other for minor.

I have implemented `sortBy` anyway.

Reply:
If that's the case I apologize for misreading that